compiler/rustc_codegen_cranelift/example/mini_core_hello_world.rs

```diff
@@ -585,6 +585,7 @@ pub enum E2<X> {
     V4,
 }
 
+#[allow(unreachable_patterns)]
 fn check_niche_behavior() {
     if let E1::V2 { .. } = (E1::V1 { f: true }) {
         intrinsics::abort();
```

Comment:
Does the MIR for this function keep both matches? If not, this wouldn't test what it is intended to test anymore.

Reply:
Yes: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=b88ecbaaeea020ab1055a977170749a0. MIR lowering is affected by this, but only in the case of an enum with one non-empty variant and a bunch of empty variants.